app/models/node.rb

```diff
@@ -164,6 +172,65 @@ def delete_with_history!(new_node, user)
     end
   end
 
+  def self.delete_with_history_bulk!(nodes, changeset, if_unused = false)
+    node_hash = nodes.collect { |node| [node.id, node] }.to_h
+    skipped = {}
+    Node.transaction do
+      # check if node ids exists
+      node_ids = node_hash.keys
+      old_nodes = Node.select("id, version, visible").where(:id => node_ids).lock
+      raise OSM::APIBadUserInput, "Node not exist. id: " + (node_ids - old_nodes.collect(&:id)).join(", ") unless node_ids.length == old_nodes.length
```

Comment:
Non-existing object should return HTTP 404 (Not found) instead of HTTP 400 (Bad Request)